small fixes and feature enhancements

binnichtaktiv opened the pull request:
I just found this great project last week and it works great! I noticed a few small things to fix tho.

- 'About' in menu bar doesn't do anything. I fixed it so it now opens this repo in default browser.
- When using 'Open backup', file instead of folder picker is used. This doesn't make sense imo because you just want to select a backup folder and not a file with file type 'iTunes backup' (I understand what it does but folder picker is better in this case I think)
- Added a simple popup that appears when exporting process is completed. Before you didn't really know when the export was complete because there was no feedback (when I first used this tool I just checked if the folder still grows in size). Now there's a popup that just says finished exporting x files.
- All dialogs are now centered. Before they appeared in random places, sometimes behind the main screen or on another monitor.

I am not a pro in Java so if you find anything to fix pls let me know.

Tested on Debian 13 and Windows 11.

binnichtaktiv committed:
add functionality to 'About' button in menu bar

binnichtaktiv committed:
add functionality to 'About' button in menu bar

binnichtaktiv committed:
replace file with folder picker & add popup

binnichtaktiv committed:
replace file with folder picker & add popup

binnichtaktiv committed:
add progress for search & popup for files tab

binnichtaktiv committed:
wth

binnichtaktiv committed:
added missing Platform.runLater

im not 100% sure if you actually need it but idk how else it should work...

binnichtaktiv committed:
popups are now centered & removed a few comments

MaxiHuHe04 commented:
Hi @binnichtaktiv,
With the file picker, I tried to make clear that it has to be a folder with a Manifest.plist/.db in there. But yes, maybe a folder picker and an error message when it does not have one makes more sense.
I have noticed two things when testing your changes on my Windows 10 PC:
- I get an `IllegalStateException: Cannot set owner once stage has been set visible` at `ui.Dialogs.centerOnOwner`(Dialogs.java:39) when I export files. It works, but the progress window does not have an icon.
- The success dialog is not centered for me, it appears at the top left of my screen.

In the future, I would prefer fewer unnecessary changes to the code formatting or e.g. removing `this.` as it makes the diffs a bit cluttered.

What would you think about moving the export task to a separate class that could be used from both file search tab and files tab?  This would prevent some code duplication and I think handling files that already exist would make sense for both of them.


Thank you for your contribution!

binnichtaktiv commented:
Thanks for your reply. My bad I just learned it that way so I changed it and removed this haha.

Right now I don't have time to do anything new. I did this when I had free time but idk when I will have time for this.

If you have time you maybe _could_ do it? You can see the changes I made and you are probably way better than me in Java... I could close this pr and create a new one with only the `about` and `file picker fix` and you maybe do the popup the way you want it. It took quite a few hours for the changes I made because its the first time I actually did something with UI and much logic

Just a question / suggestion.
